feat(SPV-1248) add domain check for paymail entry

cain4chain opened the pull request:
- feat(paymail) add checkPaymailDomain function for domain validation 
- feat(error) add error definition to engine code for invalid domain
- chore(paymailCreateAddress) add logic for domain validation check



# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

cain4chain committed:
feat(SPV-1242) add domain check for paymail entry

feat(paymail) add checkPaymailDomain function for domain validation
feat(error) add error definition to engine code for invalid domain
chore(paymailCreateAddress) add logic for domain validation check

chris-4chain commented on actions/admin/paymail_addresses_old.go:
```diff
@@ -163,6 +164,15 @@ func paymailCreateAddressOld(c *gin.Context, _ *reqctx.AdminContext) {
 		opts = append(opts, engine.WithMetadatas(requestBody.Metadata))
 	}
 
+	config := reqctx.AppConfig(c)
+	if config.Paymail.DomainValidationEnabled {
+		_, actualDomain, _ := paymail.SanitizePaymail(requestBody.Address)
+		if !checkPaymailDomain(actualDomain, config.Paymail.Domains) {
+			spverrors.ErrorResponse(c, spverrors.ErrInvalidDomain, logger)
+			return
+		}
+	}
```
This is duplicated code.
But since we plan to remove the "old" handlers soon (I believe 🙈), it is totally fine to me 👍

dorzepowski commented on actions/admin/paymail_addresses.go:
```diff
@@ -130,6 +131,15 @@ func paymailCreateAddress(c *gin.Context, _ *reqctx.AdminContext) {
+	config := reqctx.AppConfig(c)
+	if config.Paymail.DomainValidationEnabled {
+		_, actualDomain, _ := paymail.SanitizePaymail(requestBody.Address)
+		if !checkPaymailDomain(actualDomain, config.Paymail.Domains) {
```
Maybe we could simply use `slices.Contains`
```suggestion
		if !slices.Contains(config.Paymail.Domains, actualDomain) {
```

cainnusti replied:
ok I will add this and test the change locally.

dorzepowski commented on actions/admin/paymail_addresses.go:
```diff
@@ -130,6 +131,15 @@ func paymailCreateAddress(c *gin.Context, _ *reqctx.AdminContext) {
 		opts = append(opts, engine.WithMetadatas(requestBody.Metadata))
 	}
 
+	config := reqctx.AppConfig(c)
```
IMHO PR is a good opportunity to write some endpoint tests for create paymail address endpoint.

chris-4chain committed:
Merge branch 'main' into feat(SPV-1248)-add-domain-check-for-paymail-entry

chris-4chain committed:
feat(SPV-1248): suggestion how to make endpoint tests

cain4chain committed:
feat(1248) : Add slices method replace custom function

- Chore (checkPaymailDomain) Remove checkPaymailDomain funciton
- Chore (paymailCreateAddress) Use "slices" method for checking Domain values

cainnusti committed:
feat(1248) : Add slices method replace custom function

- Chore (checkPaymailDomain) Remove checkPaymailDomain funciton
- Chore (paymailCreateAddress) Use "slices" method for checking Domain values

cain4chain committed:
Merge branch 'feat(SPV-1248)-add-domain-check-for-paymail-entry' of https://github.com/bitcoin-sv/spv-wallet into feat(SPV-1248)-add-domain-check-for-paymail-entry

cain4chain committed:
feat(SPV-1248) add tests to remove the paymail

feat(test) - add test to remove paymail as admin and check that the paymail was removed. try to remove paymail as non admin should fail

cain4chain committed:
feat(SPV-1248) add paymail tests

chore(lint-styles): fix linting issues

cain4chain committed:
Merge branch 'main' into feat(SPV-1248)-add-domain-check-for-paymail-entry
